### ISSUE TITLE
upgrade from 2.2.3 to 2.2.4 in any MC apps `Target Projects  *` is null

### ISSUE THREAD

**Issue by Negashev** [rancher/rancher#20698]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
Upgrade from 2.2.3 to 2.2.4

**Result:**

In Multi-Cluster Apps target project selector lost
![Annotation 2019-06-06 104610](https://user-images.githubusercontent.com/4653589/59015955-dbe8ca00-8848-11e9-8b81-6ff39432a288.jpg)

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): server v2.2.4, UI v2.2.70
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Infrastructure Provider
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM openstack
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:35:57 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by Negashev** [comment]
Maybe I will need to roll back migrations, since during the update I had 3 nodes and when the update was updated, the first 2.2.4 node rebooted and probably did not have time to roll up the schema updates

**Comment by Negashev** [comment]
Is it possible to re-roll migration and check the schema?

**Comment by davidnuzik** [comment]
@Negashev I wasn't able to reproduce but you might possibly be running into https://github.com/rancher/rancher/issues/19586 I notice that you saw v2.2.4 at the bottom left and that the UI version is reporting correctly, but does this still happen when you clear the browser cache (or Chrome Incognito for example)? I am not seeing this problem with a very similar environmental configuration.

**Comment by Negashev** [comment]
@davidnuzik Yes, I tried incognito and completely cleared the cache, also checked from the phone, and through Microsoft EDGE

**Comment by davidnuzik** [comment]
Thanks @Negashev I primarily do frontend so I'm looking to see if we can get someone else to assist and answer your questions. It is indeed possible there is an issue with the schema or similar issue based on that node that restarted during upgrade.

**Comment by Negashev** [comment]
@davidnuzik Anything to solve this problem!
I can show a screen from slack call (@negash)

**Comment by westlywright** [comment]
@Negashev if you view the MC App in the API (on apps page select app action menu -> view in api) does the `targets` parameter have data resembling the following?

```
{
"appId": "<appId>",
"projectId": "<projectId>",
"state": "deploying",
"type": "/v3/schemas/target"
}
```

**Comment by Negashev** [comment]
@westlywright It's not empty for all projects
![Screenshot (3)](https://user-images.githubusercontent.com/4653589/59063409-aed00200-88b0-11e9-8dff-c8b5ed778868.png)

**Comment by Negashev** [comment]
deploy from cli `rancher mcapps ....` work

**Comment by soumyalj** [comment]
Observed this issue in Master build from June 11
![Screen Shot 2019-06-12 at 4 45 21 PM](https://user-images.githubusercontent.com/18536626/59393701-883b1c80-8d31-11e9-8165-5a1c278e1170.png)

**Comment by davidnuzik** [comment]
Refer to #20857 for backport issue (v2.2.5) that version is validated. Next up is check in master (v2.3) but this is lower priority for me since this is being released later.

**Comment by davidnuzik** [comment]
I tested with Master now and can confirm the issue is resolved in latest master as well.

**Replication, Version: v2.2.3**
I created orphaned projects by creating eks clusters and deleting before ready. (Verify in kubectl there are ns left behind)
I replicated the Target projects issue:

![image](https://user-images.githubusercontent.com/45179589/59529179-2693c300-8e95-11e9-9f28-c288980ea840.png)

**Validation, Version: master**
I upgraded to master and we now filter out orphaned projects so the Target drop down now works correctly. I confirmed post-upgrade it works whether or not we have clusters up yet. When I bring additional clusters up the list updates as expected and I smoke tested multi-cluster apps and they all deployed correctly. There are no longer any web console errors as well. HTTP/API/Apps look correct and are created in all designated areas (projects I specify).

![image](https://user-images.githubusercontent.com/45179589/59529370-b0439080-8e95-11e9-8bd6-25c22037d5fc.png)

![image](https://user-images.githubusercontent.com/45179589/59529413-d1a47c80-8e95-11e9-8de0-c46fc7bae9ec.png)
